### ISSUE TITLE
SBEMU hangs on Dell Latitude D530

### ISSUE THREAD

**Issue by bert003** [crazii/SBEMU#38]
When I boot I get the following messages and the system stops booting there (refer to the attached image):

![image](https://github.com/crazii/SBEMU/assets/20399702/8e54eaf8-0612-4db1-aa33-1e7b9b680d98)

I grouped all the files in a directory called SBEMU and changed paths into CONFIG.SYS and AUTOEXEC.BAT accordingly.

What could be the issue please?

**Comment by bert003** [comment]
Did anyone have a similar issue please?

**Comment by tiagotarifa** [comment]
> Did anyone have a similar issue please?

Yes, I do!
Same problem a D530 too. 

I was trying to compile sbemu with some "printfs" to see where exactly it's freeze, but I couldn't find a way to compile this. So I gave up.

Maybe one day with time I can try this again.

(I tried a new version today: UserBuild_2023.12.25_19-41. Same problem)

**Comment by crazii** [comment]
I'll add some extra msg to help locating the problem.

**Comment by crazii** [comment]
> > Did anyone have a similar issue please?
> 
> Yes, I do! Same problem here in a D530 too.
> 
> I was trying to compile sbemu with some "printfs" to see where exactly it's freeze, but I couldn't find a way to compile it. So I gave up.
> 
> Maybe one day with time I can try this again.
> 
> (I tried a new version today: UserBuild_2023.12.25_19-41. Same problem)

Have you read the latest readme?  you can build on Linux, Windows(WSL/MinGW)

**Comment by tiagotarifa** [comment]
Today I made a new test with **UserBuild_2024.01.14_11-11**. 

Now it's not freeze any more when it load. However, **there is no sound**. I tried on Doom 1.9 shareware and BioMenace.

A curious thing I discovered is when I try to change output with `sbemu.exe /O0` (headphones) it's made some kind of "pop" noise on headphones, like trying to activate the sound card, and then it's give an error.

![tela1](https://github.com/crazii/SBEMU/assets/17298749/34394af1-8658-44c8-a4b9-24bb520f89a4)
![tela2](https://github.com/crazii/SBEMU/assets/17298749/243bf8f0-b1bf-4906-89c3-34600c79674c)

**Additional information**
- Parallel port and IEE1904 are disabled;
- Tests were made booting this freedos image by PXE boot;
- Changing IRQ port 7 to 5 make no difference;
- Others options like reset (/R) and change volume (/VOL9) was tried too;

**Comment by hjnijlunsing** [comment]
- Does it work with other games? With my Intel HDA; especially Dooms seems problematic

- Does it work with https://github.com/Baron-von-Riedesel/VSBHDA/releases/tag/v1 ?
  My Intel HDA behaves differently with the two alternatives. And doom does work with this fork

- Did the BIOS assign an IRQ? (https://github.com/Baron-von-Riedesel/VSBHDA/issues/1)

- Does it work with MPXPlay?

**Comment by tiagotarifa** [comment]
> * Does it work with other games? With my Intel HDA; especially Dooms seems problematic

I tried BioMenace. Do you have other suggestion?

**Comment by hjnijlunsing** [comment]
Epic Pinball works with my Intel HDA on SBEMU, can you check?

**Comment by tiagotarifa** [comment]
> Epic Pinball works with my Intel HDA on SBEMU, can you check?

Same problem, no sound. I tried to change sbemu options like IRQ and VOL and nothing changed.

Another curious thing: Before sbemu load the pcspeaker (emulated by soundcard) works. It's stop to work after it's load.

**Comment by hjnijlunsing** [comment]
The PC Speaker; I had exactly the same issue with VSBHDA: https://github.com/Baron-von-Riedesel/VSBHDA/issues/2
You did try VSBHDA, MPXPlay and confirm your HDA has an IRQ assigned? (as per https://github.com/Baron-von-Riedesel/VSBHDA/issues/1)

**Comment by tiagotarifa** [comment]
> You did try VSBHDA, MPXPlay and confirm your HDA has an IRQ assigned?
Yes, I did. No sound ether. ( https://github.com/Baron-von-Riedesel/VSBHDA/issues/3#issuecomment-1891012728)

> MPXPlay
I tried now. It recognise the sound card, show playing a wav file but still no sound.

> and confirm your HDA has an IRQ assigned
pciinfo gave me this:
0 1b  0       4  3  0  2  8086/284b   9  multimedia controller (HDA)
I didn't got you "EMPTY" issue.

**Other test that I made** 
- On BIOS, I disabled USB, Parallel port, COM ports, wifi, bluetooth, network without PXE, network at all, usb "emulator" (for msdos boot from usb). The problem still the same;

@hjnijlunsing, thank you for all your support and information.

**Comment by hjnijlunsing** [comment]
Ok, crazii is looking at the intel hda code at the moment (https://github.com/crazii/SBEMU/issues/13). 
Let's wait if a potential fix might also work for your case.

Meanwhile if you need any help in compiling per directions of crazii let me know; I could compile the project for you if required.

**Comment by tiagotarifa** [comment]
@crazii , great news! (and bad news too)

**GREAT NEWS**
Using version **UserBuild_2024.02.03_17-10** works but it's need some steps:
1. Run with these parameters: 
`sbemu /VOL9`
2. Then change the output:
`sbemu /O0`

This way doesn't work:
`sbemu /VOL9 /O0`

_Sound and FM works. OPL3 doesn't_

**BAD NEWS**
Games crashed after run for a while. Games tested:
- Doom: Freezes;
- Epic Pinball: Freeze after leave Android table;
- Boom: Crashed when I tried to run asetup.exe to configure the sound;

These same games doesn't crash if I play without load sbemu and it's dependencies.

On Linux, the sound card description is this:
Intel Corporation 82801H (ICH8 family) HD Audio Controller (rev 02)

Some pictures of the problem and more information about this sound card on dmesg:
![new_IMG_20240204_201713](https://github.com/crazii/SBEMU/assets/17298749/5ce12a49-3526-48a7-af16-1b59da3ba35a)
![new_IMG_20240204_201825](https://github.com/crazii/SBEMU/assets/17298749/3b14e335-a468-4521-ae0e-3a1ba718e904)
![new_IMG_20240204_202712](https://github.com/crazii/SBEMU/assets/17298749/9215ff88-26a2-4fe1-b5c7-a5600868554c)

**Comment by crazii** [comment]
Thanks for the report , Plz test with the UserBuild_2024.02.04_23-49 to see if there's any changes.

**Comment by crazii** [comment]
> 1. Load sbemu as usual. I did this way:
>    `sbemu /VOL9`
> 2. Then change the output:
>    `sbemu /O0`
> 
> This way doesn't work: `sbemu /VOL9 /O0`
> 
> but, if I run `sbemu /VOL9 /O0` and then `sbemu /O1` works too.

can you test with the second run with /R  (reset sound card driver)?
I believe we missed something in handling you card. but dunno what, plz be patient that more time is needed for this problem.

**Comment by tiagotarifa** [comment]
> Thanks for the report , Plz test with the UserBuild_2024.02.04_23-49 to see if there's any changes.
>...
> can you test with the second run with /R (reset sound card driver)?

I tested and It has the same behaviour. 

> I believe we missed something in handling you card. but dunno what, plz be patient that more time is needed for this problem.

Don't worry. I just mark you because it's is a great achievement, in my opinion. 
I appreciate your work!
Thank you!
